### PR TITLE
Change of access to some components of SDC to make them available for…

### PR DESCRIPTION
**Pull request by khyativyasargus**
… custom components

**IMPORTANT: All PRs must be linked to an issue (except for extremely trivial and straightforward changes).**

Fixes [#2023](https://github.com/google/android-fhir/issues/2023)

**Description**
Clear and concise code change description. 

**Alternative(s) considered**
Have you considered any alternatives? And if so, why have you chosen the approach in this PR?

**Type**
Bug fix

**Screenshots (if applicable)**

**Checklist**
- [x] I have read and acknowledged the [Code of conduct](https://github.com/google/android-fhir/blob/master/docs/CODE_OF_CONDUCT.md).
- [x] I have read the [Contributing](https://github.com/google/android-fhir/wiki/Contributing) page.
- [x] I have signed the Google [Individual CLA](https://cla.developers.google.com/about/google-individual), or I am covered by my company's [Corporate CLA](https://cla.developers.google.com/about/google-corporate ).
- [x] I have discussed my proposed solution with code owners in the linked issue(s) and we have agreed upon the general approach.
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the style guide of this project.
- [x] I have run `./gradlew check` and `./gradlew connectedCheck` to test my changes locally.
- [x] I have built and run the demo app(s) to verify my change fixes the issue and/or does not break the demo app(s).

### CODE REVIEW

**Review by jingtang10 — APPROVED** [review]

File: datacapture/src/main/java/com/google/android/fhir/datacapture/extensions/MoreQuestionnaireItemComponents.kt [hunk 1]
```diff
@@ -154,7 +154,7 @@ internal fun Questionnaire.QuestionnaireItemComponent.isReferencedBy(
   }
 
 // Item control code, or null
-internal val Questionnaire.QuestionnaireItemComponent.itemControl: ItemControlTypes?
+val Questionnaire.QuestionnaireItemComponent.itemControl: ItemControlTypes?
```
Comment (jingtang10) [inline datacapture/src/main/java/com/google/android/fhir/datacapture/extensions/MoreQuestionnaireItemComponents.kt:157]: rewrite the doc:

```suggestion
/**
 * The [ItemControlTypes] of the questionnaire item if it is specified by the item control extension, or `null`.
 *
 * See http://hl7.org/fhir/R4/extension-questionnaire-itemcontrol.html.
 */
val Questionnaire.QuestionnaireItemComponent.itemControl: ItemControlTypes?
```

File: datacapture/src/main/java/com/google/android/fhir/datacapture/extensions/MoreQuestionnaireItemComponents.kt [hunk 2]
```diff
@@ -173,7 +173,7 @@ internal val Questionnaire.QuestionnaireItemComponent.itemControl: ItemControlTy
     return ItemControlTypes.values().firstOrNull { it.extensionCode == code }
   }
 
-internal enum class ChoiceOrientationTypes(val extensionCode: String) {
+enum class ChoiceOrientationTypes(val extensionCode: String) {
```
Comment (jingtang10) [inline datacapture/src/main/java/com/google/android/fhir/datacapture/extensions/MoreQuestionnaireItemComponents.kt:176]:
```suggestion
/**
 * The desired orientation for the list of choices.
 *
 * See http://hl7.org/fhir/R4/extension-questionnaire-choiceorientation.html.
 */
enum class ChoiceOrientationTypes(val extensionCode: String) {
```

File: datacapture/src/main/java/com/google/android/fhir/datacapture/views/GroupHeaderView.kt [hunk 3]
```diff
@@ -28,8 +28,7 @@ import com.google.android.fhir.datacapture.extensions.localizedInstructionsSpann
 import com.google.android.fhir.datacapture.extensions.localizedPrefixSpanned
 import com.google.android.fhir.datacapture.extensions.updateTextAndVisibility
 
-internal class GroupHeaderView(context: Context, attrs: AttributeSet?) :
-  LinearLayout(context, attrs) {
+class GroupHeaderView(context: Context, attrs: AttributeSet?) : LinearLayout(context, attrs) {
```
Comment (jingtang10) [inline datacapture/src/main/java/com/google/android/fhir/datacapture/views/GroupHeaderView.kt:31]: add a comment here

**Review by jingtang10 — APPROVED** [review]

File: datacapture/src/main/java/com/google/android/fhir/datacapture/views/GroupHeaderView.kt (hunk 3 above)
Comment (khyativyasargus) [reply datacapture/src/main/java/com/google/android/fhir/datacapture/views/GroupHeaderView.kt:31]: > add a comment here

/**
 * Generic view for the prefix, question, and hint of a header of group of questions [QuestionnaireViewHolderType.GROUP]
 */

File: datacapture/src/main/java/com/google/android/fhir/datacapture/views/GroupHeaderView.kt [hunk 4]
```diff
@@ -21,15 +21,18 @@ import android.util.AttributeSet
...
 import com.google.android.fhir.datacapture.extensions.initHelpViews
 import com.google.android.fhir.datacapture.extensions.localizedInstructionsSpanned
 import com.google.android.fhir.datacapture.extensions.localizedPrefixSpanned
 import com.google.android.fhir.datacapture.extensions.updateTextAndVisibility
 
-internal class GroupHeaderView(context: Context, attrs: AttributeSet?) :
-  LinearLayout(context, attrs) {
+/**
+ * Generic view for the prefix, question, and hint of a header of group of questions [QuestionnaireViewHolderType.GROUP]
+ */
```
Comment (jingtang10) [inline datacapture/src/main/java/com/google/android/fhir/datacapture/views/GroupHeaderView.kt:34]:
```suggestion
 * Generic view for the prefix, question, and hint as the header of a group using a view holder of type [QuestionnaireViewHolderType.GROUP].
 */
```

**Review by jingtang10 — APPROVED** [review]

**Pull request merged**